assets/js/components/common/UserCan.jsx

```diff
@@ -23,7 +23,8 @@ class UserCan extends Component {
   }
 
   render() {
-    if (this.userCan()) return this.props.children
+    if (this.props.type == "not" && !this.userCan()) return this.props.children
```

Comment:
i'd recommend breaking it into 2 component, `<UserCan>` and `<UserCannot>` because it'll be more apparent when looking through code what's going on, and also, we can just have separate render methods with their own logic instead of switching on `props.type` here

Reply:
trick will be making sure not to duplicate the rules logic